selinux: Relabel files if and only if the policy version changed

branto1 opened the pull request:
Currently, the ceph files are being relabelled every time the package is
rebuilt. Fix this by checking the policy versions and relabel the files
only if the policy actually changed (different policy version was
detected).

liewegas commented:
lgtm.. @sidhax ?

mbroz commented:
@branto1 two comments:
 - it requires awk, this should be in base install (no specific require needed) but perhaps this applies only to RH distros. Is it correct?

- /usr/sbin/load_policy is now not called if the policy version is the same (but patch says it skips only relabel).
I think that load_policy is not needed anyway, semodule should load policy already. Just remove it?

branto1 commented:
OK, I've changed the patch to run load_policy if selinux is enabled. AFAIK, the call is necessary as semodule -n -i won't reload policy in kernel and that is by design -- otherwise, the command would fail on systems where SELinux is disabled in kernel.

As for awk, it would be nice if one of the suse guys could comment, here?

liewegas commented:
@osynge ping.. any issue with awk dependency?

mbroz commented:
@liewegas We can also add explicit Require: awk here (despite it is redundant on Fedora).

Anyway, this PR should be merged together with #5596 (which is even more important for the release).

branto1 commented:
I merged the #5596 and rebased this one on top of it. Also, I added gawk to post script requirements. We actually need this even in fedora as it is not a build-time but install-time dependency where there is pretty much no such thing as guaranteed installed packages.
